pkg/genesis/vesting_accounts.go

```diff
@@ -45,8 +45,9 @@ func NewGenesisDelayedVestingAccount(
 
 	balance = banktypes.Balance{
 		Address: address,
-		Coins:   initUnlockedCoins.Add(vestingBalance...),
+		Coins:   vestingBalance,
 	}
+	vestingBalance = vestingBalance.Sub(initUnlockedCoins...)
```

Comment:
[Suggestion, optional] `Sub` may cause panic if it results in a negative value; it could be worth mentioning the possibility of panic in the `NewGenesisDelayedVestingAccount` description. Ditto other modified functions in this PR.